### PR TITLE
[FEATURE] Add scp transfer method

### PR DESCRIPTION
**Pull request by sabbelasichon**
* **Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

* **What is the current behavior?** (You can also link to an open issue here)
Until now there is no scp transfer strategy

* **What is the new behavior (if this is a feature change)?**
We have a new transfer strategy called scp

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

### CODE REVIEW

File: src/Task/Transfer/ScpTask.php [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        // To prevent issues remove all tar.gz in it before
+        $this->shell->executeOrSimulate(sprintf('rm -rf %s/*.tar.gz', $localPackagePath), $localhost, $deployment);
+        // Create tarball locally in the workspace directory
+        $this->shell->executeOrSimulate(sprintf('cd %1$s/; tar %3$s -zcf %2$s -C %1$s .', $localPackagePath, $fileName, $this->getExcludes($options)), $localhost, $deployment, false, false);
+
+        // Transfer tarball to target server
+        $this->shell->executeOrSimulate(sprintf('scp %s/%s %s', $localPackagePath, $fileName, $destinationArgument), $localhost, $deployment);
+
+        // Extract tarball on server in release path
+        $this->shell->executeOrSimulate(sprintf('tar -zxvf %s/%s -C %s', $remoteTransferPath, $fileName, $releasePath), $node, $deployment);
```
Comment (simonschaufi) [inline src/Task/Transfer/ScpTask.php]: do we need the `v` parameter?

File: src/Task/Transfer/ScpTask.php [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        $localPackagePath = preg_replace('/\s+/', '\ ', $localPackagePath);
+        $destinationArgument = preg_replace('/\s+/', '\ ', $destinationArgument);
+
+        // Create a localhost node and create tarball on it
+        $localhost = new Node('localhost');
+        $localhost->onLocalhost();
+        // To prevent issues remove all tar.gz in it before
+        $this->shell->executeOrSimulate(sprintf('rm -rf %s/*.tar.gz', $localPackagePath), $localhost, $deployment);
+        // Create tarball locally in the workspace directory
+        $this->shell->executeOrSimulate(sprintf('cd %1$s/; tar %3$s -zcf %2$s -C %1$s .', $localPackagePath, $fileName, $this->getExcludes($options)), $localhost, $deployment, false, false);
```
Comment (simonschaufi) [inline src/Task/Transfer/ScpTask.php]: I get an error here on my Ubuntu 18 machine (surprisingly not always but in 99%):
```
(localhost): "cd /home/xxx/.surf/workspace/live/xxx/ && tar --exclude=".git" -zcf 20200202191056.tar.gz -C /home/xxx/.surf/workspace/live/xxx ."

Got exception "Command returned non-zero return code: 1" but rollback disabled. Stopping.
```

Comment (simonschaufi) [reply src/Task/Transfer/ScpTask.php]: When I extracted the tar.gz I noticed that the file itself is also included.

File: src/Task/Transfer/ScpTask.php (hunk 1 above)
Comment (sabbelasichon) [reply src/Task/Transfer/ScpTask.php]: @simonschaufi No, don´t think so. Maybe we could make it configurable one day

**Pull request merged**